### ISSUE TITLE
In Python3 calling _execute_command_with_vdom() raises TypeError in pyFG

### ISSUE THREAD

**Issue by ebeahan** [napalm-automation/napalm-fortios#40]
### Description of Issue/Question

While trying out Python3 support on the `develop` branch, I encountered an exception in `pyFG` when the `napalm-fortios` driver calls `_execute_command_with_vdom()`. Appears to only impact Python3; Python2 works as expected.

This issue is to document behavior for `napalm-fortios`. Will investigate root cause and open a separate issue/PR in the ``pyFG`` repo.

### Steps to Reproduce the Issue

```
$ cat test.py
#!/usr/bin/env python

from napalm_base import get_network_driver
import pprint

pp = pprint.PrettyPrinter(indent=4)

fortios_config = dict(hostname='192.168.1.245', username='admin', password='password')
driver = get_network_driver('fortios')
with driver(**fortios_config) as fortios:
    pp.pprint(fortios.get_firewall_policies())
```

Python2 works as expected:

```
$ python2 test.py
{   '1': [   {   u'action': u'permit',
                 u'byte_hits': -1,
                 u'dst_zone': u'wan1',
                 u'enabled': False,
                 u'id': u'1',
                 u'l3_dst': u'all',
                 u'l3_src': u'all',
                 u'log': u'False',
                 u'packet_hits': -1,
                 u'position': 1,
                 u'schedule': u'always',
                 u'service': u'ALL',
etc.....
````

Python3 returns the below Traceback.

```
$ python3 test.py
````

### Error Traceback
```
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    pp.pprint(fortios.get_firewall_policies())
  File "/Users/ebeahan/Envs/napalm-fortios/lib/python3.5/site-packages/napalm_base/base.py", line 60, in __exit__
    self.__raise_clean_exception(exc_type, exc_value, exc_traceback)
  File "/Users/ebeahan/Envs/napalm-fortios/lib/python3.5/site-packages/napalm_base/base.py", line 82, in __raise_clean_exception
    raise exc_value
  File "test.py", line 11, in <module>
    pp.pprint(fortios.get_firewall_policies())
  File "/Users/ebeahan/dev/napalm-fortios/napalm_fortios/fortios.py", line 271, in get_firewall_policies
    cmd = self._execute_command_with_vdom('show firewall policy')
  File "/Users/ebeahan/dev/napalm-fortios/napalm_fortios/fortios.py", line 74, in _execute_command_with_vdom
    return self.device.execute_command(command)
  File "/Users/ebeahan/Envs/napalm-fortios/lib/python3.5/site-packages/pyFG/fortios.py", line 137, in execute_command
    output += o
TypeError: Can't convert 'int' object to str implicitly
```

**Comment by ktbyers** [comment]
Yes, it looks like we should be able to just change the following lines in pyFG/fortios.py

```
133         for e in error_chan.read():
134             error += e
135 
136         for o in output_chan.read():
137             output += o
```

To:

```
133         for e in error_chan.read():
134             error = error + str(e)
135 
136         for o in output_chan.read():
137             output = output + str(o)
```

On the face of it...I don't see why we are receiving an `int` here, however (I would have thought that `o` would always be a string).

**Comment by ebeahan** [comment]
When I was digging at this yesterday the first character being `.read()` off the channel for my device was a `'J'` (device hostname starting with 'J') in python2 and `74` in python3 which is the Unicode code point for 'J'. I made an experimental edit like your suggestion but switching `str` with `chr` and that fixed the issue, but that wouldn't be backwards compatible with python2.

**Comment by ktbyers** [comment]
What version of Paramiko are you on?

**Comment by ebeahan** [comment]
`paramiko==2.1.1`

**Comment by ktbyers** [comment]
Okay...we need to handle the unicode on the .read() i.e. it always needs to come back as unicode. I should be able to do something similar to what I did in Netmiko.

**Comment by ktbyers** [comment]
@ebeahan Okay, I pushed a change to my pyfg repository. Can you test it?

https://github.com/ktbyers/pyfg

You should test this with both py2 and py3.

It would be good if we could run the pyfg/unittests using both Py2 and Py3, but this would probably require a virtual device (which I don't have).

**Comment by ebeahan** [comment]
@ktbyers Change appears to have fixed the original issue - thanks! I did run into one other issue which was the couple of times `.iteritems()` was called in `forticonfig.py` would thrown an exception under python3. After I went in and changed `.iteritems()` for `items()` I didn't encounter any other issues.

https://github.com/ktbyers/pyfg/blob/master/pyFG/forticonfig.py#L190

**Comment by ktbyers** [comment]
@ebeahan Okay, let me change those and submit a PR back to pyfg...strange I missed those (let me double check my pylama config on pyfg and try to make sure I am not missing others).

**Comment by ktbyers** [comment]
@ebeahan Submitted this PR here...

https://github.com/spotify/pyfg/pull/16


Can you test my pyFG branch, I want to make sure pyFG (with fixes) is working properly using PY3.

My branch is here:

https://github.com/ktbyers/pyfg

**Comment by ebeahan** [comment]
@ktbyers will do

**Comment by ebeahan** [comment]
@ktbyers Everything looks good while testing with your branch. Once your PR gets merged in for pyFG I think we should be set.

**Comment by ebeahan** [comment]
Left a comment on @ktbyers PR to pyFG to try and get some forward movement. We're back waiting for an update to pyFG in order to release Python3 support for napalm-fortios.

cc @dbarrosop

**Comment by ebeahan** [comment]
Resolved with 0.4.0 release.
